edx/edx-exams
# feat: update exam attempt status when starting LTI proctoring launch

**JIRA:** [MST-1842](https://2u-internal.atlassian.net/browse/MST-1842)

**Description:** 

This commits adds behavior to update a learner's exam attempt status to the "download_software_clicked" status when the learner starts the LTI proctoring launch via the start_proctoring view.

**Author concerns:** 

* The `start_proctoring` view only handles `GET` requests. This doesn't feel as appropriate given that it changes the `ExamAttempt` state - it's a side effecting GET. However, the URL doesn't really uniquely define a resource (i.e. the `ExamAttempt`). A `PUT` would be more appropriate, but it would make it more difficult to manually test without wiring up the learning MFE to `edx-exams` because you can no longer just put the URL in the browser. What do you think? I can log a ticket to update the request handling once we have a better way to test the entire LTI flow via the learning MFE.

**Dependencies:** None.

**Installation instructions:** Please follow the instructions in [Local Development and LTI Configuration](https://2u-internal.atlassian.net/wiki/spaces/PT/pages/256737327/Local+Development+and+LTI+Configuration) to set up your environment.

**Testing instructions:**

1. Create an `ExamAttempt` with the `created` status.
2. Visit `http://localhost:18740/lti/start_proctoring/<attempt_id>`.
3. Verify that the `ExamAttempt` has transitioned to the `download_software_clicked` state and that you see the corresponding logs in the console.

**Merge checklist:**

**Post merge:**
